node/core/dispute-coordinator/src/real/ordering/mod.rs

```diff
@@ -207,14 +235,94 @@ impl OrderingProvider {
 	}
 }
 
+async fn send_message_fatal<Sender, Response>(
+	sender: &mut Sender,
+	message: impl Into<AllMessages>,
+	receiver: oneshot::Receiver<std::result::Result<Response, ChainApiError>>,
+) -> FatalResult<Response>
+where
+	Sender: SubsystemSender,
+{
+	sender.send_message(message.into()).await;
+
+	receiver
+		.await
+		.map_err(|_| Fatal::ChainApiSenderDropped)?
+		.map_err(Fatal::ChainApi)
```

Comment:
Considering that this returns chain api errors, the function should probably also only accept chain api messages.

Reply:
I've thought about it but the message itself includes the oneshot sender, so it's not clonable.
Possible approaches are:
- Introduce new `Fatal` errors for each request type.
- Or supply a context via debug formatting the message beforehand.

Reply:
We are talking about different things: What I was referring to was to not pass in a generic [impl Into<AllMessages>](https://github.com/paritytech/polkadot/pull/4508/files/218d70480067d5be8d5d22a7f10fd2acc0b63ddd#diff-bde8b4a248f2d59bf94e264b0df2d786997a408d490a28bee6f9e15e8f9ab4d1R240), but a concrete `ChainApiMessage` instead (because the errors are specialized for that).

Reply:
Oh, sure, the comment refers to line 251 which looks like `Fatal::ChainApi` should accept a message.